pkg/ipcache/ipcache.go

```diff
@@ -189,6 +192,25 @@ const (
 	Delete CacheModification = "Delete"
 )
 
+func keyToIP(key string) (net.IP, error) {
+	requiredPrefix := fmt.Sprintf("%s/", path.Join(IPIdentitiesPath, AddressSpace))
+	if !strings.HasPrefix(key, requiredPrefix) {
+		return nil, fmt.Errorf("Found invalid key %s outside of prefix %s", key, IPIdentitiesPath)
+	}
+
+	suffix := strings.TrimPrefix(key, requiredPrefix)
+	if suffix[0] == '/' {
+		suffix = suffix[1:]
+	}
+
+	parsedIP := net.ParseIP(suffix)
+	if parsedIP == nil {
+		return nil, fmt.Errorf("unable to parse IP from key %s", key)
```

Comment:
Add the suffix here and not the whole key, we want to know what was attempted to be parsed.

Reply:
Fixed.